daemon/main.go

```diff
@@ -650,6 +653,7 @@ func initEnv(cmd *cobra.Command) {
 			scopedLog.WithError(err).Fatal("Unable to restore agent assets")
 		}
 	}
+	option.Config.MaxControllerInterval = viper.GetInt(option.MaxCtrlIntervalName)
```

Comment:
Why do we need this line?

Check for negative numbers

Reply:
I double-checked and the new option was not functional without this line (at least for the bound variable).

There's currently value checking in the usage of this but I agreed that the checking should occur over here instead. Will fix it up